### PR TITLE
Image storage

### LINKED ISSUE #3: Decide on image storage implementation
11/6 
Assigned to Mandeep, Daniel, Ira

### LINKED ISSUE #4: Save images to local storage
Crucial to upload, download, and edit features

### LINKED ISSUE #10: Upload Image back-end functionality
Workload includes:

1. Relating HTML form to javascript
2. Storing images to local storage
3. Type-checking image and handling error cases

### PR DESCRIPTION
**Pull request by mchera24**
Workload includes:

- Developed the image storage functionality
- Handles the case when an empty image is stored, when a wrong file extension file is uploaded, and the general upload of a correct image file.

Testing:

- I tested this functionality through vscode's live server extension
- I tried uploading an empty image and it correctly alerts the user and does not add anything to local storage
- I tried uploading multiple images and it correctly adds them to the array data structure in local storage
- I tried uploading a PDF document and it correctly alerts the user that the file type is invalid and does not add it to local storage

Potential Improvements:

- Thinking of other potential edge cases

### CODE REVIEW

File: source/assets/scripts/upload.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+window.addEventListener('DOMContentLoaded', init);
+
+function init()
+{
+    const upload_form = document.getElementById('upload-form');
+    const imageList = [];
```
Comment (nkarter02) [inline source/assets/scripts/upload.js:6]: stay consistent with either under_scores or camelCase, could refer to styling.md ADR

File: source/assets/scripts/upload.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        //stores the allowed image extensions
+        let allowedFileTypes = ['jpg', 'png', 'jpeg'];
+        //event listener for when the form is submitted and the page loads
+        file.addEventListener('load', () => 
+        {
+            //as long as local storage exists/is supported on the browser
+            if(localStorage)
+            {
+                //holds the value of if the file extension is correct
+                var isCorrect = new Boolean(false);
```
Comment (nkarter02) [inline source/assets/scripts/upload.js]: not sure if we should use the 'var' keyword

File: source/assets/scripts/upload.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                    //when it detects the correct file extension set the bool var to true to indicate that
+                    if(allowedFileTypes[count] == file_extension)
+                    {
+                        isCorrect = true;
+                    }
+                }
+                //in the case that the file extension is invalid, let the user know and return
+                if(isCorrect != true)
+                {
+                    alert('Invalid file type!');
```
Comment (nkarter02) [inline source/assets/scripts/upload.js]: Alert user that the only allowed file types are jpeg, jpg, and png

File: source/assets/scripts/upload.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                    //add a new array with the image uploaded to local storage
+                    imageList.push(newImg);
+                    window.localStorage.setItem('Image Container', JSON.stringify(imageList));
+                }
+                else
+                {
+                    //add the new image object to the existing array
+                    image_list.push(newImg);
+                    window.localStorage.setItem('Image Container', JSON.stringify(image_list));
+                }
```
Comment (nkarter02) [inline source/assets/scripts/upload.js]: I could be misreading this but isn't the 'if' block the same as the 'else' block

**Review by nkarter02 — CHANGES_REQUESTED** [review]
I made a few comments on upload.js, although those are just on code style and not functionality, (I will test it on my device soon)

File: source/assets/scripts/upload.js (hunk 1 above)
Comment (mchera24) [reply source/assets/scripts/upload.js:6]: done

File: source/assets/scripts/upload.js (hunk 2 above)
Comment (mchera24) [reply source/assets/scripts/upload.js]: done

File: source/assets/scripts/upload.js (hunk 3 above)
Comment (mchera24) [reply source/assets/scripts/upload.js]: done

File: source/assets/scripts/upload.js (hunk 4 above)
Comment (mchera24) [reply source/assets/scripts/upload.js]: done - changed variable names to make it more clear

**Comment by nkarter02** [comment]
If scrolled down on the page, if you highlight the "Eighteam Photoshop" text in the top left, the page moves upwards. Not sure if that is intended

**Comment by nkarter02** [comment]
> If scrolled down on the page, if you highlight the "Eighteam Photoshop" text in the top left, the page moves upwards. Not sure if that is intended

Turns out it can be if you highlight any of the text on the top area

**Comment by nkarter02** [comment]
Could change text from "Get started by inserting an image below:" to "Upload more images:" after the user properly uploads their first image

**Review by nkarter02 — COMMENTED** [review]
Variable names are now camelCase, good job!

**Review by nkarter02 — APPROVED** [review]

**Pull request merged**